tests/test_inheritance.rs

```diff
@@ -208,6 +209,38 @@ mod inheriting_native_type {
             r#"dict_sub[0] = 1; assert dict_sub[0] == 1; assert dict_sub._name == "Hello :)""#
         );
     }
+
+    #[pyclass(extends=PyException)]
+    #[derive(Debug)]
+    struct CustomException {
+        #[pyo3(get)]
+        context: &'static str,
+    }
+
+    #[pymethods]
+    impl CustomException {
+        #[new]
+        fn new() -> Self {
+            CustomException {
+                context: "Hello :)",
+            }
+        }
+    }
+
+    #[test]
+    fn custom_exception() {
+        Python::with_gil(|py| {
+            let cls = py.get_type::<CustomException>();
+            let dict = [("cls", cls)].into_py_dict(py);
+            let res = py.run(
+            "e = cls('hello'); assert str(e) == 'hello'; assert e.context == 'Hello :)'; raise e",
```

Comment:
I guess it's a bit redundant, but do you also want to check that the exception can be caught in Python code?

Reply:
I don't see the harm in doing so; it's more comprehensive adding it 👍